### PR TITLE
Feat 188 paginate results in status page

### LINKED ISSUE #188: Paginate status page table
**Is your feature request related to a problem? Please describe.**
As a user, I want to be able to page through the table of documents in the status page.

**Describe the solution you'd like**
A generic Pagination component if one does not exist, and the ability to page through results in the status page using the Pagination component. 

**Describe alternatives you've considered**
N/A

**Additional context**
N/A

### PR DESCRIPTION
**Pull request by jhackshaw-dds**
Thanks for submitting a pull request! Below are a few things you can do to help us more quickly review your changes.

## Checklist
I have…
- [x] summarized below my changes and noted which issues (if any) this pull request fixes or addresses.
- [x] thoroughly outlined below the steps necessary to test my changes.
- [x] attached screenshots illustrating relevant behavior before and after my changes.
~~[ ] review and update SSAG documentation if needed.~~

## Summary of Changes
This pull request…
- closes #188 
- create pagination component
- add types for paginated api responses
- add pagination plugin to table component
- add pagination component to status page
- add page to api queries for documents

## Testing
To verify the changes proposed in this pull request…
1. Run unit tests
2. Test pagination in development environment

## Screenshots
image

image

image

### CODE REVIEW

File: frontend/src/components/Paginator/PaginatorButton.tsx [hunk 1]
```diff
@@ -0,0 +1,35 @@
+import React from "react";
+import classNames from "classnames";
+import { Button } from "solo-uswds";
+import classes from "./Paginator.module.scss";
+
+interface PaginatorButtonProps {
```
Comment (SCasarottoDDS) [inline frontend/src/components/Paginator/PaginatorButton.tsx]: Extend base button props.

File: frontend/src/solo-types/apiDoc.ts [hunk 2]
```diff
@@ -174,6 +174,13 @@ export const defaultApiDoc: ApiDocument = {
   sdn: "M3030012341234"
 };
 
+export const defaultApiResponse: PaginatedApiResponse<ApiDocument[]> = {
+  results: [defaultApiDoc],
+  count: 100,
+  next: 1,
+  previous: 2
+};
+
```
Comment (SCasarottoDDS) [inline frontend/src/solo-types/apiDoc.ts:183]: Have you used an api before that returns like this? I would expect it to be `currentPage`, `pageSize` and `results`. Don't care either way just surprised.

**Review by SCasarottoDDS — CHANGES_REQUESTED** [review]

File: frontend/src/solo-types/apiDoc.ts (hunk 2 above)
Comment (jhackshaw-dds) [reply frontend/src/solo-types/apiDoc.ts:183]: This is how Django Rest Framework default pagination structures results.

**Review by SCasarottoDDS — APPROVED** [review]

**Comment by SCasarottoDDS** [comment]
image
Format error?

**Pull request merged**